src/Stratis.Bitcoin/EventBus/CoreEvents/Peer/PeerMessageReceived.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Net;
+using Stratis.Bitcoin.P2P.Protocol;
+
+namespace Stratis.Bitcoin.EventBus.CoreEvents
+{
+    /// <summary>
+    /// A peer message has been received and parsed
+    /// </summary>
+    /// <seealso cref="Stratis.Bitcoin.EventBus.EventBase" />
+    public class PeerMessageReceived : PeerEventBase
+    {
+        public Message Message { get; }
```

Comment:
The same here, move `Message` to be base. It can be null :)

Reply:
I don't agree on this: PeerConnected/PeerConnectionAttempt/PeerConnectionAttemptFailed/PeerDisconnected doesn't have a concept of Message so having that property, even if null, is not a good design because the model communicate that we may have e Message, that's incorrect